### PR TITLE
Unblock Core CI

### PR DESCRIPTION
**Pull request by antkmsft**
This test started to fail around Jan 9th, it fails consistently, but on Linux only.
Further investigations should be made, but CI needs to be unblocked sooner, and this is what this PR does.

More info:
I compiled with Curl transport on Windows, it does not repro.
CI failure logs:
first,
```
[  INFO ] /mnt/vss/_work/1/s/sdk/core/azure-core/test/ut/transport_policy_options.cpp:549:: Test https://www.example.com/
231: [2023-01-10T19:34:34.6835347Z] DEBUG : [CURL Transport Adapter]: Creating a new session.
231: [2023-01-10T19:34:34.6835697Z] DEBUG : [CURL Transport Adapter]: Spawn new connection.
231: [2023-01-10T19:35:38.7378656Z] DEBUG : [CURL Transport Adapter]: No Host in request headers. Adding it
231: [2023-01-10T19:35:38.7379044Z] DEBUG : [CURL Transport Adapter]: No content-length in headers. Adding it
231: [2023-01-10T19:35:38.7379310Z] DEBUG : [CURL Transport Adapter]: Send request without payload
231: [2023-01-10T19:35:38.7384886Z] DEBUG : [CURL Transport Adapter]: Parse server response
231: [2023-01-10T19:35:38.7459252Z] DEBUG : [CURL Transport Adapter]: Request completed. Moving response out of session and session to response.
231: [2023-01-10T19:35:38.7459741Z] DEBUG : Moving connection to pool...
231: [2023-01-10T19:35:38.7460087Z] DEBUG : Clean thread running. Won't start a new one.
```

then,
```
231:   Actual: it throws Azure::Core::Http::TransportException with description "Fail to get a new connection for: https://www.example.com. Timeout was reached".
```

Place in the test code where it fails:
https://github.com/Azure/azure-sdk-for-cpp/blob/01af6783b1ffa4be1717fd8d413e874dcee5620d/sdk/core/azure-core/test/ut/transport_policy_options.cpp#L578-L579

### CODE REVIEW

File: sdk/core/azure-core/test/ut/transport_policy_options.cpp [hunk 1]
```diff
@@ -524,7 +524,13 @@ namespace Azure { namespace Core { namespace Test {
 #endif
   }
 
+// This test started to fail consistently in CI on Linux.
+// It should be fixed, but to unblock CI, there is a workaround below.
+#if !defined(AZ_PLATFORM_LINUX)
   TEST_F(TransportAdapterOptions, MultipleCrlOperations)
+#else
+  TEST_F(TransportAdapterOptions, MultipleCrlOperations_LIVEONLY_)
+#endif
```
Comment (ahsonkhan) [inline sdk/core/azure-core/test/ut/transport_policy_options.cpp:533]: Please file an issue to track the CI failure. Is it rare and intermittent (where a re-run succeeds), or does it happen quite frequently?

Comment (antkmsft) [reply sdk/core/azure-core/test/ut/transport_policy_options.cpp:533]: I saw it yesterday and since then, it is consistent, does not fail on rerun.

Comment (ahsonkhan) [reply sdk/core/azure-core/test/ut/transport_policy_options.cpp]: If the issue is with example.com, it is surprising that it only fails on linux.

**Review by ahsonkhan — APPROVED** [review]

File: sdk/core/azure-core/test/ut/transport_policy_options.cpp (hunk 1 above)
Comment (antkmsft) [reply sdk/core/azure-core/test/ut/transport_policy_options.cpp]: #4237

Comment (antkmsft) [reply sdk/core/azure-core/test/ut/transport_policy_options.cpp]: I think it is an issue with the connection pool, not with any site.

**Pull request merged**